File: bysj/src/main/java/com/union/bangbang/bysj/ui/user/startup/StartUpActivity.java
package com.union.bangbang.bysj.ui.user.startup;

import android.os.Bundle;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.union.bangbang.bysj.BR;
import com.union.bangbang.bysj.R;
import com.union.bangbang.bysj.base.arouter.RouteNavigation;
import com.union.bangbang.bysj.databinding.ActivityStartupBinding;

import me.goldze.mvvmhabit.base.BaseActivity;

import static com.union.bangbang.bysj.base.arouter.RouteNavigation.Router.login;

/**
 * @name toto
 * @anthor bangbang QQ:740090077
 * @time 2019/1/5 3:56 PM
 * 只有编译器可能不骗你。
 */
@Route(path = RouteNavigation.Router.startup)
public class StartUpActivity extends BaseActivity<ActivityStartupBinding,StartUpModel> {
    @Override
    public int initContentView(Bundle bundle) {
        return R.layout.activity_startup;
    }
    @Override
    public int initVariableId() {
        return BR.vm;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        RouteNavigation.Navigation(login);
    }

}

File: todokotlin/src/main/java/com/union/bangbang/todokotlin/ui/startup/StartUpActivity.kt
package com.union.bangbang.todokotlin.ui.startup

import android.os.Bundle
import com.union.bangbang.todokotlin.R
import com.union.bangbang.todokotlin.base.activity.BaseActivity
import com.union.bangbang.todokotlin.base.model.BaseModel
import com.union.bangbang.todokotlin.base.utils.ArouterUtil
import com.union.bangbang.todokotlin.dagger.module.ActivityModule.Companion.home_page
import com.union.bangbang.todokotlin.databinding.ActivityStartupBinding
import io.reactivex.functions.Consumer
import javax.inject.Inject


class StartUpActivity : BaseActivity<ActivityStartupBinding>() {
    override fun initViewModel(): BaseModel? = viewModel
    @Inject
    lateinit var viewModel: StartUpModel;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setImmersive()
        viewModel.refresh()
        viewModel.startTiming(Consumer {
            if (it) {
                ArouterUtil.navigation(home_page)
                finish()
            }
        }, time = 1000)
    }

    override fun getLayoutId(): Int = R.layout.activity_startup
}
